mlir/lib/Dialect/Bufferization/IR/BufferizationOps.cpp

```diff
@@ -542,25 +542,35 @@ bool MaterializeInDestinationOp::bufferizesToMemoryRead(
 
 bool MaterializeInDestinationOp::bufferizesToMemoryWrite(
     OpOperand &opOperand, const AnalysisState &state) {
-  return &opOperand == &getDestMutable();
+  return isa<TensorType>(getDest().getType()) &&
```

Comment:
Couldn't we make the `isa` an assert because the interface method's description says that it will never be called on non-tensor operands?